homeassistant/components/bmw_connected_drive/config_flow.py

```diff
@@ -104,6 +104,17 @@ async def async_step_account_options(
     ) -> FlowResult:
         """Handle the initial step."""
         if user_input is not None:
+            # Manually update & reload the config entry after options change.
+            # Required as each successful login will store the latest refresh_token
+            # using async_update_entry, which would otherwise trigger a full reload
+            # if the options would be refreshed using a listener.
+            if entry := self.hass.config_entries.async_get_entry(self.handler):
```

Comment:
We already have `self.config_entry`.

Reply:
Oops, thank you. Took me too long to figure out how to best reload so I totally missed this!